src/exchange/exchange.go

```diff
@@ -207,9 +218,20 @@ func (s *Exchange) Shutdown() {
 func (s *Exchange) saveIncomingDeposit(dv scanner.Deposit) (DepositInfo, error) {
 	log := s.log.WithField("deposit", dv)
 
-	log.Info("Received bitcoin deposit")
+	var rate string
+	switch dv.CoinType {
+	case scanner.CoinTypeBTC:
+		log.Info("Received bitcoin deposit")
+		rate = s.cfg.Rate
+	case scanner.CoinTypeETH:
+		log.Info("Received ethcoin deposit")
+		rate = s.cfg.EthRate
+	default:
+		log.Error("unsupport cointype %s", dv.CoinType)
```

Comment:
`log.WithError(err).Error()`

`dv.CoinType` is already in the log because of `log := s.log.WithField("deposit", dv)`